shoop/core/api/orders.py

```diff
@@ -45,6 +48,49 @@ class Meta:
         model = Order
 
 
+class DefaultPaymentSerializer(serializers.Serializer):
+    payment_identifier = serializers.CharField(max_length=50)
+    description = serializers.CharField(max_length=256, allow_null=True)
+
+
+class PaymentSerializer(DefaultPaymentSerializer):
+    amount_value = serializers.DecimalField(decimal_places=9, max_digits=36)
+    currency = serializers.CharField(max_length=10)
+
+
 class OrderViewSet(ModelViewSet):
     serializer_class = OrderSerializer
     queryset = Order.objects.all()
+
+    @detail_route(methods=['post'])
+    def create_payment(self, request, pk=None):
+        order = self.get_object()
+        serializer = PaymentSerializer(data=request.data)
+        if serializer.is_valid():
+            data = serializer.validated_data
+            order.create_payment(
+                Money(data["amount_value"], data["currency"]),
+                data["payment_identifier"],
+                data.get("description", "")
+            )
+            return Response({'status': 'payment created'})
+        else:
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    @detail_route(methods=['post'])
+    def set_fully_paid(self, request, pk=None):
+        order = self.get_object()
+        if order.is_paid():
+            return Response({"status": "order is already fully paid"})
+        serializer = DefaultPaymentSerializer(data=request.data)
+        if serializer.is_valid():
```

Comment:
Ok, thanks, will check these ASAP. And I think that we can use the same serializer for both.